controllers/nutanixcluster_controller_test.go

```diff
@@ -161,10 +143,8 @@ func TestNutanixClusterReconciler(t *testing.T) {
 				g.Expect(result.RequeueAfter).To(BeZero())
 				g.Expect(result.Requeue).To(BeFalse())
 			})
-			It("should not error and not requeue if failure domains are configured and cluster is Ready", func() {
-				ntnxCluster.Spec.FailureDomains = []infrav1.NutanixFailureDomain{
-					fd1,
-				}
+			It("should not error and not requeue if controlPlaneFailureDomains are configured and cluster is Ready", func() {
+				ntnxCluster.Spec.ControlPlaneFailureDomains = []corev1.LocalObjectReference{fdRef}
```

Comment:
we should add a separate test for controlplanefailuredomains instead of modifying existing ones